private/model/api/api.go

```diff
@@ -704,6 +796,72 @@ func (a *API) ServiceGoCode() string {
 	return code
 }
 
+// ServiceTestGoCode generates service client specific unit-tests
+func (a *API) ServiceTestGoCode() string {
+	// At this time we only need to generate client tests for external config resolution
+	if !a.HasExternalServiceConfigFields() {
```

Comment:
Nit should be `Client`TestGoCode :) or maybe `APIClientTestGoCode` if we're using that naming.

what do you think about putting all of these templates, and behavior into its own go file. e.g. `client_config.go` similar to `smoke.go` and `examples.go`.